qualtran/bloqs/chemistry/trotter/hubbard/hubbard.ipynb

```diff
@@ -421,6 +421,254 @@
+   },
+   "outputs": [],
+   "source": [
+    "from qualtran.drawing import show_bloqs\n",
+    "show_bloqs([hopping_tile_hwp],\n",
+    "           ['`hopping_tile_hwp`'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d3a484f4",
+   "metadata": {
+    "cq.autogen": "HoppingTileHWP.call_graph.md"
+   },
+   "source": [
+    "### Call Graph"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a6c1a69a",
+   "metadata": {
+    "cq.autogen": "HoppingTileHWP.call_graph.py"
+   },
+   "outputs": [],
+   "source": [
+    "from qualtran.resource_counting.generalizers import ignore_split_join\n",
+    "hopping_tile_hwp_g, hopping_tile_hwp_sigma = hopping_tile_hwp.call_graph(max_depth=1, generalizer=ignore_split_join)\n",
+    "show_call_graph(hopping_tile_hwp_g)\n",
+    "show_counts_sigma(hopping_tile_hwp_sigma)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f6ae394e",
+   "metadata": {
+    "cq.autogen": "InteractionHWP.bloq_doc.md"
+   },
+   "source": [
+    "## `InteractionHWP`\n",
+    "Bloq implementing the hubbard U part of the hamiltonian using Hamming weight phasing.\n",
+    "\n",
+    "Specifically:\n",
+    "$$\n",
+    "    U_I = e^{i t H_I}\n",
+    "$$\n",
+    "which can be implemented using equal angle single-qubit Z rotations.\n",
+    "\n",
+    "Each interaction term can be implemented using a e^{iZZ} gate, which\n",
+    "decomposes into a single Rz gate flanked by cliffords. There are L^2\n",
+    "equal angle rotations in total all of which may be applied in parallel using HWP.\n",
+    "\n",
+    "#### Parameters\n",
+    " - `length`: Lattice length L.\n",
+    " - `angle`: The rotation angle for unitary.\n",
+    " - `hubb_u`: The hubbard U parameter.\n",
+    " - `eps`: The precision for single qubit rotations. \n",
+    "\n",
+    "#### Registers\n",
+    " - `system`: The system register of size 2 `length`. \n",
+    "\n",
+    "#### References\n",
+    " - [Early fault-tolerant simulations of the Hubbard model](     https://arxiv.org/abs/2012.09238) Eq. page 13 paragraph 1, and page     14 paragraph 3 right column. The apply 2 batches of $L^2/2$ rotations.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dc032232",
+   "metadata": {
+    "cq.autogen": "InteractionHWP.bloq_doc.py"
+   },
+   "outputs": [],
+   "source": [
+    "from qualtran.bloqs.chemistry.trotter.hubbard.interaction import InteractionHWP"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "14a37bd0",
+   "metadata": {
+    "cq.autogen": "InteractionHWP.example_instances.md"
+   },
+   "source": [
+    "### Example Instances"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c8e4267a",
+   "metadata": {
+    "cq.autogen": "InteractionHWP.interaction_hwp"
+   },
+   "outputs": [],
+   "source": [
+    "length = 8\n",
+    "angle = 0.5\n",
+    "hubb_u = 4.0\n",
+    "interaction = InteractionHWP(length, angle, hubb_u)"
```

Comment:
the variable name must match the bloq_example name

Reply:
Whoops